### DISCUSSION TITLE
adding a link to edit trashed entries to the Actions column of the section addedby TrashOperation

### DISCUSSION THREAD

**Discussion by prescriptionlifeline** [Laravel-Backpack/community-forum#1137] [Q&A (Help)] [answered]
I followed the tutorial added at https://backpackforlaravel.com/articles/tutorials/how-to-create-a-trash-deleted-section-in-backpack-crud to add a Trash/Deleted section to my Backpack CRUD. It works well BUT I'd like to be able to edit accounts from this section and idk how to do that.

As an example of a use case wherein this would be handy...  let's say that each account has a Note field where you're supposed to explain why you trashed an account. If you trash an account before you update this Note then, right now, you'd have to then restore the account, edit it, and then archive the account a second time. It'd be nice if you could just edit it straight from the Trash/Deleted section, instead.

Any ideas?

**Comment by karandatwani92** [comment]
Hey @prescriptionlifeline 

For this use case, you can choose to [Create a New Operation With a Form](https://backpackforlaravel.com/docs/6.x/crud-operations#creating-a-new-operation-with-a-form-1) to leave a comment before archive:

```
<?php

namespace App\Http\Controllers\Admin\Operations;

use Backpack\CRUD\app\Http\Controllers\Operations\Concerns\HasForm;

trait ArchiveOperation
{
    use HasForm;

    /**
     * Define which routes are needed for this operation.
     *
     * @param string $segment    Name of the current entity (singular). Used as first URL segment.
     * @param string $routeName  Prefix of the route name.
     * @param string $controller Name of the current CrudController.
     */
    protected function setupArchiveRoutes(string $segment, string $routeName, string $controller): void
    {
        $this->formRoutes(
            operationName: 'archive',
            routesHaveIdSegment: true,
            segment: $segment,
            routeName: $routeName,
            controller: $controller
        );
    }

    /**
     * Add the default settings, buttons, etc that this operation needs.
     */
    protected function setupArchiveDefaults(): void
    {
        $this->formDefaults(
            operationName: 'archive',
            buttonStack: 'line', // alternatives: top, bottom
            // buttonMeta: [
            //     'icon' => 'la la-home',
            //     'label' => 'Archive',
            //     'wrapper' => [
            //          'target' => '_blank',
            //     ],
            // ],
        );
    }

    /**
     * Method to handle the GET request and display the View with a Backpack form
     *
     * @param  int  $id
     * @return \Illuminate\Contracts\View\View
     */
    public function getArchiveForm(int $id)
    {
        $this->crud->hasAccessOrFail('archive');
        $this->crud->addField([ // Text
            'name' => 'archive',
            'label' => 'Archive',
            'type' => 'textarea',
        ]);
        return $this->formView($id);
    }

    /**
     * Method to handle the POST request and perform the operation
     *
     * @param  int  $id
     * @return array|\Illuminate\Http\RedirectResponse
     */
    public function postArchiveForm(int $id)
    {
        $this->crud->hasAccessOrFail('archive');

        return $this->formAction(id: $id, formLogic: function ($inputs, $entry) {
            // You logic goes here...
            // dd('got to ' . __METHOD__, $inputs, $entry);

            // show a success message
            \Alert::success('Entry Archived!')->flash();
        });
    }
}
```

**Reply by prescriptionlifeline** [reply]
I was _not_ asking how to add a Note before trashing a user - I was asking how to _edit_ trashed entries.

The scenario I gave was just _one_ scenario.

What if I wanted to edit the note two months after the fact? Maybe I'm a site admin that has a website wherein users can be soft deleted and one user, two months after they were trashed, sent my company threatening snail mail. I might want to add that to the note.

And what if I just want to view the data in a trashed account? In theory this could be done with `ShowOperation` but if I do that then, for non-trashed accounts, I'd have both "Preview" and "Edit" in the "Actions" column, which I quite dislike. Like if you want to Preview an account you can just Edit an account. The notion of separate "Preview" and "Edit" actions is like having two versions of Microsoft Notepad - one that opens files in readonly mode and one that opens them in write mode. Like if I want to open a file without making any changes maybe people can just trust me _not to make any changes_? And if I'm going to let "Edit" double as "Preview" in for non-trashed account then I'd just assume have a consistent UI instead of one UI for non-trashed accounts and another readonly UI for trashed accounts.

**Comment by prescriptionlifeline** [comment]
Since this question was closed without a satisfactory answer and since it doesn't appear that I have the appropriate permissions to unclose it I'm going to just re-create it.

**Comment by prescriptionlifeline** [comment, marked as answer]
I came up with a few ways to do this.

# Method # 1

Copy `setupTrashFilter()` from backpack/pro/src/Http/Controllers/Operations/Traits/TrashFilter.php to your custom CRUD and add the button:

```
CRUD::addButton('line', 'edit_trashed', 'view', 'vendor.backpack.crud.buttons.edit_trashed');
```

Then add your new button to this line:

```php
$buttons_to_remove = array_filter(Arr::pluck($this->crud->buttons(), 'name'), function ($item) {
    return !in_array($item, ['destroy', 'restore', 'trash']);
});
```

Then create resources/views/vendor/backpack/crud/buttons/edit_trashed.blade.php with the following contents:

```
   <a href="{{ url($crud->route.'/'.$entry->getKey().'/edit') }}" class="btn btn-sm btn-link">
       <i class="la la-edit"></i> Edit
   </a>
```

# Method # 2

Copy vendor\backpack\pro\resources\views\buttons\destroy.blade.php to resources\views\vendor\backpack\pro\buttons\destroy.blade.php and then add the button to that:

```
@if ($entry->trashed())
<a href="{{ url($crud->route.'/'.$entry->getKey().'/edit') }}" class="btn btn-sm btn-link">
    <i class="la la-edit"></i> Edit
</a>
@endif
```
